diesel/src/mysql/types/mod.rs

```diff
@@ -4,7 +4,34 @@ mod date_and_time;
 use mysql::{Mysql, MysqlType};
 use std::error::Error as StdError;
 use std::io::Write;
-use types::{ToSql, IsNull, FromSql, HasSqlType};
+use types::{ToSql, IsNull, NotNull, FromSql, HasSqlType};
+
+
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq, PartialOrd, Ord)] pub struct Tinyint;
+
+impl NotNull for Tinyint {}
+
+impl ToSql<Tinyint, Mysql> for i8 {
+    fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<StdError+Send+Sync>> {
+        let int_value:i32 = *self as i32;
+        <i32 as ToSql<::types::Integer, Mysql>>::to_sql(&int_value, out)
```

Comment:
I guess this could work. I would personally prefer if you used

```rust
out.write_i8(*self)                                     
    .map(|_| IsNull::No)                                
    .map_err(|e| Box::new(e) as Box<StdError+Send+Sync>)
```